### DISCUSSION TITLE
Keep extensions data

### DISCUSSION THREAD

**Discussion by ghost** [Cookie-AutoDelete/Cookie-AutoDelete#1076] [Q&A / Support] [answered]
Hi everyone!

I want to enable automatic cleaning, and purge everything that's not in the whitelist, including "Other Browsing Data Cleanup Options"
But what about extensions?
There are some that you can go to settings page and add to whitelist
(like this one, located at moz-extension://83a9c635-a5c1-417d-ba57-f1f784db5fe5/settings/settings.html)
But most of them don't open the settings page, and are only managed by right-click-dropdown-menu.

What should I do to preserve extensions data?
Or is this will not impact it?

**Comment by kennethtran93** [comment, marked as answer]
If extensions are storing data through cookies and other browsing data (normally accessed through `window.*`), then they are not storing data the recommended way.  

Extension data should be stored using the storage API provided from each browser (`storage.local.set`), which is stored separately and will persist until the extension gets removed or the entire browser profile gets deleted.  Storage API areas consists of 'managed (Company)', 'local (Your device)', and 'sync (between same-browsers/accounts)'.

If they do store data through the browsing data storage, then that data can be cleaned up at anytime by extensions or through the browser's "Clean/Clear Browsing Data" functions.

Note that only cookies are available for iteration/checking.  Extensions do not have the ability to check site's localStorage, indexedDB, etc.  We only have the ability to request specific sites to be cleaned, or all sites by time, which is what will happen if you initially enable the 'other browsing data cleanup' option in settings.

In our extension, we only set cookies for sites that do not set a cookie, extension pages do not get any cookies set from us, and thus are usually not affected by the regular cleanups.
